plugins/importexport/onix30/filter/MonographONIX30XmlFilter.inc.php

```diff
@@ -322,20 +323,20 @@ function createProductNode($doc, $submission, $publicationFormat) {
 
 		$productTitleDetailNode->appendChild($titleElementNode);
 
-		if (!$publication->getLocalizedData('prefix') || !$publication->getLocalizedData('title')) {
-			$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitleText', trim($publication->getLocalizedData('prefix') ?? $publication->getLocalizedTitle())));
+		if (!$publication->getLocalizedData('prefix', $pubLocale) || !$publication->getLocalizedData('title', $pubLocale)) {
+			$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitleText', trim($publication->getLocalizedData('prefix', $pubLocale) ?? $publication->getLocalizedTitle($pubLocale))));
 		} else {
-			if ($publication->getLocalizedData('prefix')) {
-				$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitlePrefix', $publication->getLocalizedData('prefix')));
+			if ($publication->getLocalizedData('prefix', $pubLocale)) {
+				$titleElementNode->appendChild($this->_buildTextNode($doc, 'TitlePrefix', $publication->getLocalizedData('prefix', $pubLocale)));
```

Comment:
For publication object I would always use getData instead of getLocalizedData, if we do not want to provide any string we find, e.g.: There could be no prefix in a language other than the submission langauge. getLocalizedData would then return that prefix, even if it does not fit with the actual submission language, and it actually should not be there for submission language.